### PR TITLE
[Auditbeat] Login dataset: Add event category and type

### PR DESCRIPTION
**Pull request by cwurm**
Adds `event.category: authentication` and `event.type: authentication_success` (or `authentication_failure`).

I also took the opportunity to add another unit test for a successful login.

### CODE REVIEW

File: x-pack/auditbeat/module/system/login/login_test.go [hunk 1]
```diff
@@ -58,6 +98,7 @@ func TestFailedLogins(t *testing.T) {
 	config["login.wtmp_file_pattern"] = ""
 	config["login.btmp_file_pattern"] = "../../../tests/files/btmp_ubuntu1804"
 	f := mbtest.NewReportingMetricSetV2(t, config)
+	defer f.(*MetricSet).utmpReader.bucket.DeleteBucket()
```
Comment (andrewkroh) [inline x-pack/auditbeat/module/system/login/login_test.go]: Why was this added? IIUC the data directory is removed in between tests so the whole DB gets deleted.

Comment (cwurm) [reply x-pack/auditbeat/module/system/login/login_test.go]: That's true, but for some reason that does not cause it to be re-read. Copy pasting one of the test functions causes the second one to fail, every time. I turned on debug logging and it still has the data written in the first function.

I didn't find out exactly why this happens, but I suspect it's because we use a `sync.Once` here, and the data path will never get re-resolved:

https://github.com/elastic/beats/blob/d36eb5a293ab2abc963168e9c6184c6e912f9877/auditbeat/datastore/datastore.go#L35-L47

This only really matters in tests as far as I can see, so I didn't dig deeper into it but just added the `DeleteBucket()` call.

Comment (andrewkroh) [reply x-pack/auditbeat/module/system/login/login_test.go]: Oh yeah, that's what it is. `OpenBucket` uses the singleton instance and the DB path is only ever initialized once. The singleton instance is closed when all the users (metricsets) close their buckets. But when it gets reused again it still will have the original path so our cleanup has no effect.

One option would be to have a package global function that the metricsets use to open a bucket. That fn handle could be swapped for testing purposes. We use this pattern in a few places. For example

```
var openBucket = datastore.OpenBucket
```

Then for testing swap in a different func.

```
openBucket = func(name string) (Bucket, error) {
    return datastore.New(paths.Join(testDir, "beat.db"), 0600).OpenBucket(name)
}
```

Comment (cwurm) [reply x-pack/auditbeat/module/system/login/login_test.go]: Yeah, we could do that separately. Merging this for now.
